src/Middleware/CheckTenantForMaintenanceMode.php

```diff
@@ -29,7 +29,12 @@ public function handle($request, Closure $next)
                 return $next($request);
             }
 
-            throw new MaintenanceModeException($data['time'], $data['retry'], $data['message']);
+            throw new HttpException(
+                503,
+                'Service Unavailable',
+                null,
+                isset($data['retry']) ? ['Retry-After' => $data['retry']] : []
+            );
```

Comment:
Was `MaintenanceModeException` removed?

Reply:
It's deprecated since L8, and with Symfony 6 using it it throws a TypeError.

Reply:
And the code above is how Laravel implements maintenance mode in L8+?

I'd like to avoid making a breaking change here if people depend on the old exception in some way. But if the new implementation is similar to how Laravel does this, it's probably fine.

Reply:
Yeah, I copied this from Laravel's codebase :)

Reply:
Cool, we can go with this then